opencv-glib/image.cpp

```diff
@@ -128,22 +314,33 @@ gcv_image_convert_color(GCVImage *image,
  * @image: A #GCVImage.
  * @rectangle: A #GCVRectangle to specify area.
  * @color: A #GCVColor to specify line color.
+ * @drawing_options: A #GCVDrawingOptions to specify optional parameters.
  *
- * It draws a rectangle to @rectangle area with @color color.
+ * It draws a rectangle to @rectangle area with @color color and @drawing_options options.
  *
  * Since: 1.0.0
  */
 void
 gcv_image_draw_rectangle(GCVImage *image,
                          GCVRectangle *rectangle,
-                         GCVColor *color)
+                         GCVColor *color,
+                         GCVDrawingOptions *drawing_options)
 {
   auto cv_image = gcv_matrix_get_raw(GCV_MATRIX(image));
   auto cv_rectangle = gcv_rectangle_get_raw(rectangle);
   auto cv_color = gcv_color_get_raw(color);
-  cv::rectangle(*cv_image,
-                *cv_rectangle,
-                *cv_color);
+  if (drawing_options) {
+    cv::rectangle(*cv_image,
+                  *cv_rectangle,
+                  *cv_color,
+                  gcv_drawing_options_get_thickness(drawing_options),
+                  gcv_drawing_options_get_line_type(drawing_options),
+                  gcv_drawing_options_get_shift(drawing_options));
```

Comment:
We can use `GCVDrawingOptionsPrivate` directly in this file:

```diff
diff --git a/opencv-glib/image.cpp b/opencv-glib/image.cpp
index b3e9260..3681bba 100644
--- a/opencv-glib/image.cpp
+++ b/opencv-glib/image.cpp
@@ -330,12 +330,13 @@ gcv_image_draw_rectangle(GCVImage *image,
   auto cv_rectangle = gcv_rectangle_get_raw(rectangle);
   auto cv_color = gcv_color_get_raw(color);
   if (drawing_options) {
+    auto options_priv = GCV_DRAWING_OPTIONS_GET_PRIVATE(drawing_options);
     cv::rectangle(*cv_image,
                   *cv_rectangle,
                   *cv_color,
-                  gcv_drawing_options_get_thickness(drawing_options),
-                  gcv_drawing_options_get_line_type(drawing_options),
-                  gcv_drawing_options_get_shift(drawing_options));
+                  options_priv->thickness,
+                  options_priv->line_type,
+                  options_priv->shift);
   } else {
     cv::rectangle(*cv_image,
                   *cv_rectangle,
```

Reply:
ok